site/get-started.markdown

```diff
@@ -0,0 +1,137 @@
+---
+title: Get Started
+page: get-started
+isGettingStarted: true
+---
+
+# Get started
+Welcome, new Haskeller! Read on to quickly set up your Haskell dev environment, execute your first lines of code, and get directions for further learning!
+
+## Content
+  - [Set up Haskell dev environment](#set-up-haskell-dev-environment)
+    - [GHCup: universal installer](#ghcup-universal-installer)
+    - [Editor](#editor)
+  - [Running first lines of code](#running-first-lines-of-code)
+  - [Writing your first Haskell program](#writing-your-first-haskell-program)
+  - [Join the community](#join-the-community)
+  - [Next steps](#next-steps)
+
+## Set up Haskell dev environment
+
+### GHCup: universal installer
+
+[GHCup](https://www.haskell.org/ghcup/#) is a universal installer for Haskell that will install for you everything you need to program in Haskell, and then will also help you manage those installations in the future (update, switch versions, ...).
+
+Follow instructions at [GHCup webpage](https://www.haskell.org/ghcup/#) to install GHCup. Then, use it to install the Haskell Toolchain, which consists of:
+
+1. **GHC** -> Haskell compiler. We will use it below to run our examples, but in practice, you will mostly be using a build tool like `cabal` or `Stack` to build your code, instead of `GHC` directly.
+2. **HLS** -> Haskell Language Server -> You won't use this directly, instead your code editor will use it in the background to provide you with a great experience while editing Haskell code.
+3. **cabal** -> Haskell build tool -> You will use this to structure your Haskell projects, build them, run them, define dependencies, ...
+4. **Stack** -> Haskell build tool -> alternative to `cabal`
```

Comment:
Notice that `stack` needs some special configuration to integrate with `hls`. Actually, `ghcup` warn you about this when installing `stack`. If newcommer skip the configuration step, `hls` will complain about "could not fin ghc" or something like that. I've already seen a few reddit/SO question about this, so maybe it worths to point this out. 

The ghcup output looks like this:
```
[ Info  ] verifying digest of: stack-2.9.1-linux-x86_64-static.tar.gz
[ Info  ] Unpacking: stack-2.9.1-linux-x86_64-static.tar.gz to /home/<username>/.ghcup/tmp/ghcup-e994793785bef45f
[ Info  ] Installing stack
[ Info  ] Stack manages GHC versions internally by default. In order to make it use ghcup installed
[ ...   ] GHC versions you can run the following commands:
[ ...   ]   stack config set install-ghc false --global
[ ...   ]   stack config set system-ghc  true  --global
[ ...   ] 
[ ...   ] On windows, you may find the following config options useful too:
[ ...   ]   skip-msys, extra-path, extra-include-dirs, extra-lib-dirs
[ ...   ] 
[ ...   ] Also check out: https://docs.haskellstack.org/en/stable/yaml_configuration
[ ...   ] 
[ ...   ] !!! Additionally, you should upgrade stack only through ghcup and not use 'stack upgrade' !!!
[ ...   ] 
Success
```

Also, new `stack` version allows for instalation hooks, but I don't know it `ghcup` uses that by default yet (I think it is comming) @hasufell might help here.

Reply:
I could use some help with figuring out what we write about this. @hasufell , any immediate opinion on this?